lib/susedistribution.pm

```diff
@@ -193,7 +193,7 @@ sub script_sudo($$) {
     }
     type_string "clear\n";    # poo#13710
     type_string "su -c \'$prog\'\n";
-    handle_password_prompt;
+    handle_password_prompt unless ($testapi::username eq 'root');
```

Comment:
I would rather add this inside handle_password_promt function, it should not harm when you use this function in right place and improve stability in other cases

Reply:
Will do that, but semantically it made more sense to me to leave it here. Basically I read it as "if I am root, there's no password prompt to handle so I don't call handle_password_prompt()"

Reply:
OK, tried moving this to the handle_password_prompt() method and it breaks other tests, so I have moved it back to the script_sudo() method.

Reply:
Sure , also I agree with your comment regarding semantic , so let's treat this as resolved